### PR TITLE
`Development`: Don't create participant entries for course-wide channels

### PR DESCRIPTION
**Pull request by lennart-keller**
#### General

- [ ] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [ ] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://ls1intum.github.io/Artemis/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://ls1intum.github.io/Artemis/dev/development-process/#naming-conventions-for-github-pull-requests).
#### Server
- [x] **Important**: I implemented the changes with a very good performance and prevented too many (unnecessary) database calls.
- [x] I followed the [coding and design guidelines](https://ls1intum.github.io/Artemis/dev/guidelines/server/).
- [x] I added multiple integration tests (Spring) related to the features (with a high test coverage).
- [x] I added pre-authorization annotations according to the [guidelines](https://ls1intum.github.io/Artemis/dev/guidelines/server/#rest-endpoint-best-practices-for-authorization) and checked the course groups for all new REST Calls (security).
- [x] I documented the Java code using JavaDoc style.

### Motivation and Context

With the introduction of course-wide channels in #7021, it is not necessary any more to create participant entries for every course member, when course-default, lecture or exercise channels are created.

### Description

The code, which is responsible for creating these participant entries is deleted.

### Steps for Testing

Prerequisites:
- 1 Admin
- 1 Student

1. Create a new course
2. Add the student to the course
3. Create a lecture and some exercises
4. Confirm that the student can see the channels

#### Performance Review
- [ ] I (as a reviewer) confirm that the client changes (in particular related to REST calls and UI responsiveness) are implemented with a very good performance 
- [ ] I (as a reviewer) confirm that the server changes (in particular related to database calls) are implemented with a very good performance
#### Code Review
- [x] Code Review 1
- [ ] Code Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

#### Server

| Class/File | Line Coverage | Confirmation (assert/expect) |
|------------|--------------:|---------------------:|
| [FileUploadExerciseImportService.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5378/latest/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.service/FileUploadExerciseImportService.html) | 100% | ✅ |
| [ModelingExerciseImportService.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5378/latest/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.service/ModelingExerciseImportService.html) | 96% | ✅ |
| [QuizExerciseImportService.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5378/latest/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.service/QuizExerciseImportService.html) | 88% | ✅ |
| [TextExerciseImportService.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5378/latest/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.service/TextExerciseImportService.html) | 100% | ✅ |
| [ExamImportService.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5378/latest/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.service.exam/ExamImportService.html) | 99% | ✅ |
| [ChannelService.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5378/latest/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.service.metis.conversation/ChannelService.html) | 94% | ✅ |
| [ProgrammingExerciseImportBasicService.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5378/latest/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.service.programming/ProgrammingExerciseImportBasicService.html) | 100% | ✅ |
| [ProgrammingExerciseService.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5378/latest/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.service.programming/ProgrammingExerciseService.html) | 93% | ✅ |
| [CourseResource.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5378/latest/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.web.rest/CourseResource.html) | 94% | ✅ |
| [ExamResource.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5378/latest/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.web.rest/ExamResource.html) | 98% | ✅ |
| [FileUploadExerciseResource.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5378/latest/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.web.rest/FileUploadExerciseResource.html) | 100% | ✅ |
| [LectureResource.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5378/latest/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.web.rest/LectureResource.html) | 97% | ✅ |
| [ModelingExerciseResource.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5378/latest/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.web.rest/ModelingExerciseResource.html) | 98% | ✅ |
| [QuizExerciseResource.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5378/latest/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.web.rest/QuizExerciseResource.html) | 93% | ✅ |
| [TextExerciseResource.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5378/latest/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.web.rest/TextExerciseResource.html) | 99% | ✅ |
| [AdminCourseResource.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5378/latest/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.web.rest.admin/AdminCourseResource.html) | 100% | ✅ |

### Screenshots

No UI Changes

### CODE REVIEW

File: src/main/java/de/tum/in/www1/artemis/service/programming/ProgrammingExerciseService.java [hunk 1]
```diff
@@ -211,8 +210,7 @@ public ProgrammingExercise createProgrammingExercise(ProgrammingExercise program
         // Save programming exercise to prevent transient exception
         ProgrammingExercise savedProgrammingExercise = programmingExerciseRepository.save(programmingExercise);
 
-        Channel createdChannel = channelService.createExerciseChannel(savedProgrammingExercise, Optional.ofNullable(programmingExercise.getChannelName()));
-        channelService.registerUsersToChannelAsynchronously(true, savedProgrammingExercise.getCourseViaExerciseGroupOrCourseMember(), createdChannel);
+        channelService.createExerciseChannel(savedProgrammingExercise, Optional.ofNullable(programmingExercise.getChannelName()));
```
Comment (MaximilianAnzinger) [inline src/main/java/de/tum/in/www1/artemis/service/programming/ProgrammingExerciseService.java:213]: Why is this optional of nullable? Can there actually be no name?

**Review by MaximilianAnzinger — APPROVED** [review]
Code LGTM, just one question.

File: src/main/java/de/tum/in/www1/artemis/service/programming/ProgrammingExerciseService.java (hunk 1 above)
Comment (lennart-keller) [reply src/main/java/de/tum/in/www1/artemis/service/programming/ProgrammingExerciseService.java:213]: Yes, if the client doesn't send a name. Then the server will generate it from the exercise title

**Review by Kroko-fant — APPROVED** [review]
Worked as expected during a testing session on ts2

**Review by basak-akan — APPROVED** [review]
Tested manually during testing session, worked as described

File: src/test/java/de/tum/in/www1/artemis/course/CourseTestService.java [hunk 2]
```diff
@@ -548,11 +548,6 @@ public void testCreateCourseWithDefaultChannels() throws Exception {
             var channels = channelRepository.findChannelsByCourseId(course2.getId());
             assertThat(channels).hasSize(DefaultChannelType.values().length);
             channels.forEach(channel -> assertThat(Arrays.stream(DefaultChannelType.values()).map(DefaultChannelType::getName)).contains(channel.getName()));
-            // Check if newly added instructor and student was added to default channels
-            channels.forEach(channel -> {
-                var participants = conversationParticipantRepository.findConversationParticipantByConversationId(channel.getId());
-                assertThat(participants).hasSize(2); // 1 instructor and 1 student
```
Comment (nityanandaz) [inline src/test/java/de/tum/in/www1/artemis/course/CourseTestService.java:554]: But tutors and instructors are still registered?

https://github.com/ls1intum/Artemis/blob/cbf79af2126fa0e0dd8f313d538e6982933177a9/src/main/java/de/tum/in/www1/artemis/service/metis/conversation/ChannelService.java#L183

Can we assert `hasSize(1)`?

**Review by dearjasmina — APPROVED** [review]
code looks good

**Review by tobias-lippert — APPROVED** [review]
Code LGTM

File: src/test/java/de/tum/in/www1/artemis/course/CourseTestService.java (hunk 2 above)
Comment (lennart-keller) [reply src/test/java/de/tum/in/www1/artemis/course/CourseTestService.java:554]: No, this is only used for exam channels and not default course channels, so no participants are added in this case here

**Review by bassner — APPROVED** [review]

**Pull request merged**